app/models/site.rb

```diff
@@ -36,5 +36,22 @@ def generate_png
 
   validates :user_id, presence: true
   validates :url, format: /\A((http|https):\/\/)*[a-z0-9_-]{1,}\.*[a-z0-9_-]{1,}\.[a-z]{2,5}(\/)?\S*\z/i
+  validate  :check_customer_status, on: :create
+  validate  :check_customer_usage, on: :create
 
+  private
+
+  def check_customer_status
+    return if user.blank?
```

Comment:
If user is blank...ignore this as this will get caught within our `validates :user_id, presence: true` check